### PR TITLE
Remove mutebug dialog

### PR DESCRIPTION
**Pull request by ronie**
it's always been a pita for skinners to have two dialogs for displaying volume info in kodi,
the major issue is getting a decent transition between the volumbar/mutebug dialogs.

this merges the mutebug functionality into the volumebar dialog.
one less xml for skinners to code and no more odd workarounds needed to get a seamless transition.

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
\o/

**Comment by MartijnKaijser** [comment]
@mkortstiege for a code check :)

**Comment by mkortstiege** [comment]
Looks good to me. Thanks.

**Comment by ronie** [comment]
jenkins build this please

File: xbmc/guilib/GUIDialog.cpp [hunk 1]
```diff
@@ -237,3 +237,9 @@ void CGUIDialog::ResetAutoClose(void)
   if (m_autoClosing && m_active)
     m_showStartTime = CTimeUtils::GetFrameTime();
 }
+
+void CGUIDialog::CancelAutoClose(void)
+{
+  if (m_autoClosing && m_active)
```
Comment (xhaggi) [inline xbmc/guilib/GUIDialog.cpp]: No need to do the check here, just set m_autoClosing to false

File: xbmc/dialogs/GUIDialogVolumeBar.cpp [hunk 2]
```diff
@@ -27,7 +28,9 @@ CGUIDialogVolumeBar::CGUIDialogVolumeBar(void)
   : CGUIDialog(WINDOW_DIALOG_VOLUME_BAR, "DialogVolumeBar.xml", DialogModalityType::MODELESS)
 {
   m_loadType = LOAD_ON_GUI_INIT;
-  SetAutoClose(VOLUME_BAR_DISPLAY_TIME);
+
+  if (g_application.IsMuted() || g_application.GetVolume(false) <= VOLUME_MINIMUM)
+    Open();
```
Comment (xhaggi) [inline xbmc/dialogs/GUIDialogVolumeBar.cpp]: Do we really need this check + the call to Open() within the ctor?

Comment (xhaggi) [inline xbmc/dialogs/GUIDialogVolumeBar.cpp]: To open the dialog on startup we should place the check and the activation to https://github.com/xbmc/xbmc/blob/74ada972bfabc63102be7442775d75e5c4a198d1/xbmc/Application.cpp#L4181

Comment (ronie) [inline xbmc/dialogs/GUIDialogVolumeBar.cpp]: the purpose of the code is to check if the volume is muted at startup, so we can show the mute icon.
not sure if there's a place where this code would fit better?

previously the visibility of the mute icon was toggled by checking the volume constantly:
https://github.com/xbmc/xbmc/pull/8875/files#diff-740af616a0870190df119e60bb08e0edL33
which is pretty meh i think...

Comment (xhaggi) [inline xbmc/dialogs/GUIDialogVolumeBar.cpp]: >previously the visibility of the mute icon was toggled by checking the volume constantly:
https://github.com/xbmc/xbmc/pull/8875/files#diff-740af616a0870190df119e60bb08e0edL33
which is pretty meh i think...

Totally agree. see my comment above where to put the check.

**Comment by ronie** [comment]
updated as @xhaggi suggested.

**Comment by xhaggi** [comment]
one last question how do we handle volume changes e. g. mute/unmute with json-rpc? do we also send the necessary actions?

**Comment by ronie** [comment]
tested json-rpc with the Input.ExecuteAction method. works ok.

**Comment by xhaggi** [comment]
@ronie you can also use `Application.SetMute` and `Application.SetVolume` json-rpc commands.

See http://kodi.wiki/view/JSON-RPC_API/v6#Application.SetMute

Both send the desired actions except if you directly set a volume value which calls `CApplication::SetVolume`

See https://github.com/xbmc/xbmc/blob/master/xbmc/Application.cpp#L4744

Could you please verify if this will work?

**Comment by xhaggi** [comment]
Ahh sorry I missed that line https://github.com/xbmc/xbmc/blob/master/xbmc/interfaces/json-rpc/ApplicationOperations.cpp#L92 which triggers the action in any case.

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Pull request merged**